.github/workflows/test.yml

```diff
@@ -66,14 +58,46 @@ jobs:
         uses: actions/cache@v2
         with:
           path: .tox
-          key: tox-cache-${{ matrix.python-version }}-${{ matrix.package }}-${{ matrix.os }}-${{ hashFiles('tox.ini', 'dev-requirements.txt') }}-core
+          key: tox-cache-${{ env.RUN_MATRIX_COMBINATION }}-${{ hashFiles('tox.ini', 'dev-requirements.txt') }}-core
       - name: run tox
-        run: tox -f ${{ matrix.python-version }}-${{ matrix.package }}
+        run: tox -f ${{ matrix.python-version }}-${{ matrix.package }} -- --benchmark-json=${{ env.RUN_MATRIX_COMBINATION }}-benchmark.json
+      - name: Find and merge benchmarks
+        # TODO: Add at least one benchmark to every package type to remove this
+        if: matrix.package == 'core'
+        run: jq -s '.[0].benchmarks = ([.[].benchmarks] | add) | if .[0].benchmarks == null then null else .[0] end' opentelemetry-*/tests/*${{ matrix.package }}*.json > output.json
+      - name: Report on benchmark results
+        # TODO: Add at least one benchmark to every package type to remove this
+        if: matrix.package == 'core'
+        uses: rhysd/github-action-benchmark@v1
+        with:
+          name: OpenTelemetry Python Benchmark
+          tool: pytest
+          output-file-path: output.json
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          # Alert with a commit comment on possible performance regression
+          alert-threshold: 200%
+          comment-always: true
+          fail-on-alert: true
+          alert-comment-cc-users: '@${{ github.event.issue.user.login }}'
+      - name: Push benchmark result
+        # TODO: Add at least one benchmark to every package type to remove this
+        if: |
+          github.event_name != 'pull_request' &&
+          matrix.package == 'core'
+        uses: rhysd/github-action-benchmark@v1
+        with:
+          name: OpenTelemetry Python Benchmark
+          tool: pytest
+          output-file-path: output.json
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          # Make a commit on `gh-pages` with benchmarks from previous step
+          auto-push: true
+          benchmark-data-dir-path: /benchmarks
```

Comment:
Is this path unique enough? Will the workflows try to override each other?

Reply:
They won't :) I have confirmed that as long as we change the "Namespace" the action will wait for all the workflows to finish and publish them all under 1 commit! See: https://nathanielrn.github.io/github-benchmarks-example/benchmarks/index.html

![image](https://user-images.githubusercontent.com/23139455/101818159-56e30e80-3ad8-11eb-8881-88ef9dd09009.png)